### DISCUSSION TITLE
Gerber to g code - Advances in PCB with two tools of different diameter

### DISCUSSION THREAD

**Discussion by MARIOBASZ** [vlachoudis/bCNC#1737] [Ideas]
Can anyone implement or know of a Gerber File to G code translator?
I have made a lot of progress on my PCB milling fixture with two different diameter tools.
intersection, union (or solid form) and difference between forms, ok.
It remains for me to complete the pocket with a 0.10mm bit where it corresponds (intersection operation), because the 0.7mm bit does not reach there (union operation), respect the non-entry of the correct limit (difference operation).

**Comment by MARIOBASZ** [comment]
![IMG_0596](https://user-images.githubusercontent.com/11743403/184198727-0662699a-3f23-4248-b46c-2d16ef5ed1f5.jpg)
![IMG_0594](https://user-images.githubusercontent.com/11743403/184198733-e74906c5-6795-4756-869e-46b9ab8a67ce.jpg)

**Reply by MARIOBASZ** [reply]
Note: This will also be useful for other tasks, such as making posters with two tools and reducing work time.
However, it will be necessary, in cases where the profile operation generates two silhouettes in the same block, to separate them so that they are in separate blocks.
![IMG_0602](https://user-images.githubusercontent.com/11743403/184201926-cf96e1dc-7a5a-49ed-b25a-ce007532afbe.jpg)

**Comment by ashokavarthanan** [comment]
Try pcb2gcode
https://github.com/pcb2gcode/pcb2gcode/releases/tag/v2.4.0

**Reply by MARIOBASZ** [reply]
Excellent!
It could have saved me time of work, if I had found it earlier!
I will evaluate whether to continue and complete with the plugin for two bits, or keep the plugin / plugins intersection, difference solid union / exclusion (from Harvie plugins)

**Comment by m1ch** [comment]
In case it is still relevant, or someone is searching for a similar tool. [Flatcam](http://flatcam.org/) is also an excellent tool.

**Reply by ashokavarthanan** [reply]
> In case it is still relevant, or someone is searching for a similar tool. [Flatcam](http://flatcam.org/) is also an excellent tool.

[Flatcam](http://flatcam.org/) is no update. 
And its created g-code is more time waste when using Extra pass.  
problem is more Unnecessary **Z** Moves when using **Extra pass**.
and **XY** same (Unnecessary) Moving (Repeated Moving when using **Extra pass**).

when not using **Extra pass**, its an excellent tool.
